Library/Formula/icecream.rb

```diff
@@ -0,0 +1,97 @@
+class Icecream < Formula
+  homepage "https://en.opensuse.org/Icecream"
+
+  option "without-docbook2X", "Build without man page"
+  option "enable-clang-wrappers", "Use symlink wrappers for clang/clang++"
+  option "enable-clang-rewrite-includes", "Use by default Clang's -frewrite-includes option"
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "docbook2X" => [:recommended, :build]
+
+  stable do
+    url "https://github.com/icecc/icecream/archive/1.0.1.tar.gz"
+    sha1 "2c3e43d34e4cbe8da5fa49e0518fd4db4c665f81"
+  end
+
+  devel do
+    url "https://github.com/icecc/icecream/archive/1.1rc1.tar.gz"
+    version "1.1rc1"
+    sha1 "ccde08f67297122710270b440492a528df042f52"
+  end
+
+  fails_with :clang do
+    build 600
+    cause "requires gnu libstdc++"
+  end
+
+  def install
+    inreplace "configure.ac", "docbook-to-man", "docbook2man" if build.stable?
+
+    args = "--disable-debug",
+           "--disable-dependency-tracking",
+           "--disable-silent-rules",
+           "--prefix=#{prefix}"
+    args << "--without-man" if build.without? "docbook2X"
+    system "./autogen.sh"
+    system "./configure", *args
+
+    system "make", "install"
+    (prefix+'org.opensuse.icecc.plist').write iceccd_plist
+    (prefix+'org.opensuse.icecc-scheduler.plist').write scheduler_plist
+  end
+
+  def caveats; <<-EOS.undent
+    To override the toolset with icecc, add to your path:
+      #{libexec}/bin/icecc-toolset
+
+    To have launchd start the icecc daemo at login:
+      cp #{prefix}/org.opensuse.icecc.plist ~/Library/LaunchAgents/
+      launchctl load -w ~/Library/LaunchAgents/org.opensuse.icecc.plist
+    EOS
+  end
+
+  def iceccd_plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+        <key>Label</key>
+        <string>Icecc Daemon</string>
+        <key>ProgramArguments</key>
+        <array>
+        <string>#{sbin}/iceccd</string>
+        <string>-d</string>
+        </array>
+        <key>RunAtLoad</key>
+        <true/>
+    </dict>
+    </plist>
+    EOS
+  end
+
+  def scheduler_plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+        <key>Label</key>
+        <string>Icecc Scheduler</string>
+        <key>ProgramArguments</key>
+        <array>
+        <string>#{sbin}/icecc-scheduler</string>
+        <string>-d</string>
+        </array>
+        <key>RunAtLoad</key>
+        <true/>
+    </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    system "#{bin}/iceccd", "-h"
+    system "#{bin}/icecc", "--version"
+    system "#{bin}/icerun", "--version"
```

Comment:
Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!